base/strings/string.jl

```diff
@@ -296,6 +296,8 @@ end
 
 isvalid(s::String, i::Int) = checkbounds(Bool, s, i) && thisind(s, i) == i
 
+isascii(s::String) = all(i -> @inbounds(codeunit(s, i)) < 0x80, 1:sizeof(s))
```

Comment:
Should it looping over `codeunits(s)`/`1:ncodeunits(s)`, or does that not matter here?

Reply:
Code units of UTF-8 are bytes, so that's equivalent, but `ncodeunits` would be more natural.

Too bad we can't just write `all(<(0x80), codeunits(s))`.